src/routes/email/email.controller.delete-email.spec.ts

```diff
@@ -245,4 +266,51 @@ describe('Email controller delete email tests', () => {
         statusCode: 403,
       });
   });
+
+  it('returns 500 if email api throws', async () => {
+    const chain = chainBuilder().build();
+    const timestamp = jest.now();
+    const privateKey = generatePrivateKey();
+    const account = privateKeyToAccount(privateKey);
+    const accountAddress = account.address;
+    // Signer is owner of safe
+    const safe = safeBuilder()
+      .with('owners', [accountAddress])
+      // Faker generates non-checksum addresses only
+      .with('address', getAddress(faker.finance.ethereumAddress()))
+      .build();
+    const email = emailBuilder()
+      .with('account', accountAddress)
+      .with('safeAddress', safe.address)
+      .with('chainId', chain.chainId)
+      .build();
+    const message = `email-delete-${chain.chainId}-${safe.address}-${accountAddress}-${timestamp}`;
+    const signature = await account.signMessage({ message });
+    networkService.get.mockImplementation((url) => {
+      switch (url) {
+        case `${safeConfigUrl}/api/v1/chains/${chain.chainId}`:
+          return Promise.resolve({ data: chain });
+        case `${chain.transactionService}/api/v1/safes/${safe.address}`:
+          return Promise.resolve({ data: safe });
+        default:
+          return Promise.reject(new Error(`Could not match ${url}`));
+      }
+    });
+    emailDatasource.getEmail.mockResolvedValueOnce(email);
+    emailApi.deleteEmailAddress.mockRejectedValue(new Error('Some error'));
+    emailDatasource.deleteEmail.mockResolvedValueOnce();
```

Comment:
Nitpick: I think this call to `mockResolvedValueOnce` is not needed as the function `emailDatasource.deleteEmail` is not expected to be called.

Reply:
Agreed 👍 [034c228](https://github.com/safe-global/safe-client-gateway/pull/994/commits/034c2286ce2058625a214fc4b44e06d40f3be97b)